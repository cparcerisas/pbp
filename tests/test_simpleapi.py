from pbp.simpleapi import Pbp
from google.cloud.storage import Client as GsClient
import pytest


def test_required_parameters_are_given():
    pbp = Pbp()
    pbp.set_json_base_dir("/tmp/some_json_dir")
    pbp.set_global_attrs_uri("s3://example.com/g.yml")
    pbp.set_variable_attrs_uri("s3://example.com/v.yml")
    pbp.set_sensitivity(3.5)
    pbp.set_subset_to((20, 4000))
    pbp.set_download_dir("/tmp/download")
    pbp.set_output_dir("/tmp/output")
    pbp.set_output_prefix("TEST_")
    pbp.set_gs_client(GsClient.create_anonymous_client())

    pbp.check_parameters()


def test_required_parameters_are_missing():
    with pytest.raises(ValueError):
        Pbp().check_parameters()
