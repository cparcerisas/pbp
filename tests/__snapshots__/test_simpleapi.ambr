# serializer version: 1
# name: test_required_parameters_are_missing
  '''
  - json_base_dir not set
  - global_attrs_uri not set
  - variable_attrs_uri not set
  - download_dir not set
  - output_dir not set
  - subset_to not set
  - subset_to must be a tuple
  - No S3 or GS client has been set
  '''
# ---
